lib/glific/erase.ex

```diff
@@ -366,4 +409,32 @@ defmodule Glific.Erase do
       )
     end
   end
+
+  @spec send_success_notification(Organization.t()) :: :ok
```

Comment:
```suggestion
 @spec send_notification(non_neg_integer(), String.t(), String.t()) ::
          {:ok, Notification.t()} | {:error, Ecto.Changeset.t()}
  defp send_notification(org_id, message, severity) do
    Notifications.create_notification(%{
      category: "HSM template",
      message: message,
      severity: severity,
      organization_id: org_id,
      entity: %{Provider: "Gupshup"}
    })
  end
 
```

We currently have separate functions for success and failure notifications, but the logic is largely identical. This results in redundant code and makes future changes harder.

Instead, we can use one generic notification function and pass message and severity as arguments.

Reply:
For reference see this file: lib/glific/templates/template_worker.ex.  line: 94